playbook.yml

```diff
@@ -45,40 +46,34 @@
       shell: oc delete pods --all -n openshift-operator-lifecycle-manager
 
     - name: Wait for PackageManifests resource to reappear once packageserver comes back
-      action: shell oc get packagemanifests
-      register: cmd_result
-      until: cmd_result.rc == 0
+      shell: oc get packagemanifests
       retries: 20
-      failed_when: cmd_result.rc not in [0,1]
+      register: result
+      until: result is succeeded
 
     - name: Wait for svcat PackageManifest to disappear
-      action: shell oc get packagemanifests | grep svcat
-      register: cmd_result
-      until: cmd_result.rc == 1
+      shell: "[ ! $(oc get packagemanifests svcat) ]"
       retries: 20
-      failed_when: cmd_result.rc not in [0,1]
+      register: result
+      until: result is succeeded
 
     - name: Create rh-operators ConfigMap
       shell: oc create -f {{ svcat_rh_operators_configmap }}
 
     - name: Create rh-operators CatalogSource
       shell: oc create -f {{ svcat_rh_operators_catalogsource_configmap }}
 
-    - name: Get catalog-operator pod name
-      shell: oc get pods -n openshift-operator-lifecycle-manager | grep catalog | grep Running | awk '{print $1}'
-      register: cmd_result
-
     - name: Delete catalog-operator pod
-      shell: oc delete pod {{ cmd_result.stdout }} -n openshift-operator-lifecycle-manager
+      shell: oc delete pod -n openshift-operator-lifecycle-manager -l app=catalog-operator
```

Comment:
Fancy! :)

Reply:
I removed all the 'grep' :)

Reply:
Seems like a good move, nice that we can pull the output directly from the potentially failing command now. 

Can't really say I regret writing the playbook with 'grep' and pipes the first time around though... I think trying to write it without 'grep' would have stopped me from creating it in an afternoon the first time around. 

Good to see examples of how you can get around it with go templating and label queries though, so +1 for that. Trying to escape everything properly is still a bit of a task for me.

Reply:
Totally agree, down and dirty gets the job done when nothing else is getting the job done.